rs/nns/governance/src/governance.rs

```diff
@@ -2293,6 +2293,38 @@ impl Governance {
         })?
     }
 
+    /// Adopts a proposal by pretending that all potential votes were YES votes in the tally.
+    ///
+    /// Preconditions:
+    /// - the proposal with `proposal_id` already exists in `self.proposal_data`
+    #[cfg(feature = "test")]
+    pub fn adopt_proposal(&mut self, proposal_id: ProposalId) -> Result<(), GovernanceError> {
+        let now = self.env.now();
+
+        let proposal_data = self.mut_proposal_data_or_err(&proposal_id, "in adopt_proposal")?;
+
+        assert_eq!(proposal_data.status(), ProposalStatus::Open);
+
+        let Some(mut tally) = proposal_data.latest_tally else {
```

Comment:
```suggestion
        let Some(tally) = proposal_data.latest_tally.as_mut() else {
```

This is to avoid the need to `replace` later.